src/Adapter/Alias/CommandHandler/BulkUpdateAliasStatusHandler.php

```diff
@@ -0,0 +1,102 @@
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Adapter\Alias\CommandHandler;
+
+use PrestaShop\PrestaShop\Adapter\Alias\Repository\AliasRepository;
+use PrestaShop\PrestaShop\Core\Domain\AbstractBulkCommandHandler;
+use PrestaShop\PrestaShop\Core\Domain\Alias\Command\BulkUpdateAliasStatusCommand;
+use PrestaShop\PrestaShop\Core\Domain\Alias\CommandHandler\BulkUpdateAliasStatusHandlerInterface;
+use PrestaShop\PrestaShop\Core\Domain\Alias\Exception\AliasException;
+use PrestaShop\PrestaShop\Core\Domain\Alias\Exception\BulkAliasException;
+use PrestaShop\PrestaShop\Core\Domain\Alias\Exception\CannotUpdateAliasException;
+use PrestaShop\PrestaShop\Core\Domain\Alias\ValueObject\AliasId;
+use PrestaShop\PrestaShop\Core\Domain\Exception\BulkCommandExceptionInterface;
+use PrestaShop\PrestaShop\Core\Domain\Feature\Exception\BulkFeatureException;
+
+class BulkUpdateAliasStatusHandler extends AbstractBulkCommandHandler implements BulkUpdateAliasStatusHandlerInterface
+{
+    /**
+     * @var AliasRepository
+     */
+    private $aliasRepository;
+
+    /**
+     * @var bool|null
+     */
+    private $enabled;
+
+    /**
+     * @param AliasRepository $aliasRepository
+     */
+    public function __construct(AliasRepository $aliasRepository)
+    {
+        $this->aliasRepository = $aliasRepository;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function handle(BulkUpdateAliasStatusCommand $command): void
+    {
+        $this->enabled = $command->isEnabled();
+        $this->handleBulkAction($command->getAliasIds(), AliasException::class);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    protected function buildBulkException(array $caughtExceptions): BulkCommandExceptionInterface
+    {
+        return new BulkAliasException(
+            $caughtExceptions,
+            'Errors occurred during Alias bulk delete action',
+            BulkFeatureException::FAILED_BULK_DELETE
+        );
+    }
+
+    /**
+     * @param AliasId $id
+     *
+     * @return void
+     */
+    protected function handleSingleAction($id): void
+    {
+        $alias = $this->aliasRepository->get($id);
+        $alias->active = $this->enabled;
```

Comment:
ahh right, now I see that abstract handler that I've introduced is still missing something.
I think you can update it.
instead of relying on $id everywhere, it could probably rely on mixed $command.
And the supports() method could as well check if its a supported command :thinking: We can do a quick pair coding session when you are available, to avoid the feedback loop.

Reply:
ahh right :thinking: 
If we introduce an interface we would need some getIds() method which would need return type (so we would as well need to have some IdInterface) and thats quite out of scope :smile: 
maybe instead of supports we could have getIds() method and still pass the mixed $command so it would end up something like this
```
    protected function handleBulkAction($command, string $exceptionToCatch): void
    {
        foreach ($this->getIds() as $id) {
            try {
                $this->handleSingleAction($id, $command);
            } catch (Throwable $e) {
                if (!($e instanceof $exceptionToCatch)) {
                    throw $e;
                }

                $this->exceptions[] = $e;
            }
        }

        if (!empty($this->exceptions)) {
            throw $this->buildBulkException($this->exceptions);
        }
    }
```
Idk :sweat: @jolelievre ? :smile: